mockwebserver/src/test/java/mockwebserver3/MockWebServerTest.java

```diff
@@ -485,20 +486,31 @@ public void tearDown() throws Exception {
   }
 
   @Test public void http100Continue() throws Exception {
-    server.enqueue(new MockResponse().setBody("response"));
-
-    URL url = server.url("/").url();
-    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-    connection.setDoOutput(true);
-    connection.setRequestProperty("Expect", "100-Continue");
-    connection.getOutputStream().write("request".getBytes(UTF_8));
-
-    InputStream in = connection.getInputStream();
-    BufferedReader reader = new BufferedReader(new InputStreamReader(in, UTF_8));
-    assertThat(reader.readLine()).isEqualTo("response");
-
-    RecordedRequest request = server.takeRequest();
-    assertThat(request.getBody().readUtf8()).isEqualTo("request");
+    server.enqueue(new MockResponse().setSocketPolicy(SocketPolicy.EXPECT_CONTINUE));
+    RawSocketClient client = new RawSocketClient(server.getPort());
+    client.sendInUtf8(
+      "POST / HTTP/1.1\r\n" +
+      "Expect: 100-Continue\r\n" +
+      "Content-Length: 0\r\n" +
+      "\r\n");
+
+    RecordedRequest requestWithExpect = server.takeRequest();
+    assertThat(requestWithExpect.getBody().readUtf8()).isEqualTo("");
+    Thread.sleep(100); // wait for whole response
```

Comment:
I've realized that response is flushed partially so there is race condition here :( I could change `readResponseUtf8String` method to read everything until socket closes but it would disable ability to make assertions during exchanges. This ability is not used yet but will be needed. For example after changing MockWebServer so it waits for request body after Expect: 100-Continue